core/src/main/java/org/elasticsearch/index/query/Rewriteable.java

```diff
@@ -36,14 +40,76 @@
      * Rewrites the given {@link Rewriteable} into its primitive form. Rewriteables that for instance fetch resources from remote hosts or
      * can simplify / optimize itself should do their heavy lifting during {@link #rewrite(QueryRewriteContext)}. This method
      * rewrites the rewriteable until it doesn't change anymore.
+     * @param original the original rewriteable to rewrite
+     * @param context the rewrite context to use
      * @throws IOException if an {@link IOException} occurs
      */
     static <T extends Rewriteable<T>> T rewrite(T original, QueryRewriteContext context) throws IOException {
+        return rewrite(original, context, false);
+    }
+
+    /**
+     * Rewrites the given {@link Rewriteable} into its primitive form. Rewriteables that for instance fetch resources from remote hosts or
+     * can simplify / optimize itself should do their heavy lifting during {@link #rewrite(QueryRewriteContext)}. This method
```

Comment:
Do you mean `{@link #rewriteAndFetch(QueryRewriteContext)}` instead of `{@link #rewrite(QueryRewriteContext)}`? Because normal rewrite doesn't fetch any resource?